editor/src/components/canvas/commands/push-intended-bounds-and-update-groups-command.ts

```diff
@@ -90,11 +111,110 @@ function pushCommandStatePatch(
   ]
 }
 
-function getGroupUpdateCommands(
+type LocalFrameAndTarget = {
+  target: ElementPath
+  parentSize: Size
+  localFrame: LocalRectangle
+}
+
+function getUpdateResizedGroupChildrenCommands(
   editor: EditorState,
   command: PushIntendedBoundsAndUpdateGroups,
-): { statePatch: EditorStatePatch; intendedBounds: Array<CanvasFrameAndTarget> } {
-  const targets = [...command.value]
+): { updatedEditor: EditorState } {
+  const targets: Array<{
+    target: ElementPath
+    size: Size
+  }> = command.value.map((ft) => ({
+    target: ft.target,
+    size: sizeFromRectangle(ft.frame),
+  }))
+
+  // we are going to mutate this as we iterate over targets
+  let updatedLocalFrames: { [path: string]: LocalFrameAndTarget | undefined } = {}
```

Comment:
```suggestion
  let updatedLocalFrames: { [path: string]: LocalFrameAndTarget } = {}
```
Surely the `undefined` would mean you could create the slightly confusing case of a key present with the value `undefined`?

Reply:
maybe there is a better way I couldn't think of:

the original problem was that typescript said the type of  `updatedLocalFrames['garbageKey']` is `LocalFrameAndTarget` and did not enforce an existence check. 
It was combined with the other typescript behavior we observed a few weeks ago when it believes that the type is never null, and nullchecks stop being properly flow typed.

Reply:
Yeah, I wish indexing was type checked, that's why I wrote `safeIndex` for arrays.

Reply:
Could add a similar function, using the `fromField` optic feels a bit overkill.